python/tvm/driver/tvmc/compiler.py

```diff
@@ -369,6 +389,20 @@ def compile_model(
         timing_inst = PassTimingInstrument()
         instruments = [timing_inst] if instruments is None else [timing_inst] + instruments
 
+    if print_ir_before:
```

Comment:
One suggestion regarding the UI - to avoid having to create separate objects when both command line arguments are present and not to rely on user having to spell "before" or "after" correctly, maybe something like 
```
if (print_ir_before or print_ir_after)
    PassPrintingInstrument(print_before_pass_names=print_ir_before, print_after_pass_names=print_ir_after)
```
and then in the instrument implementation we can simply do 
```
...
    def run_before_pass(self, mod, pass_info):
        if pass_info.name in self.print_before_pass_names:
...
    def run_after_pass(self, mod, pass_info):
        if pass_info.name in self.print_after_pass_names:
```

I think the command line arguments would also need to default to empty lists then instead of empty strings.